HAL_ChibiOS: added RaidolinkPIX6 support

radiolinkW opened the pull request:
HAL_ChibiOS: added RaidolinkPIX6 support

Tools: added RaidolinkPIX6 bootloader

peterbarker reviewed (commented):
I've created a PR to reserve the board ID you're using in here.

Please note that RadioLink still owe me access to source code per ArduPilot's GPLv3.

Having bought a MiniPix, running custom firmware from RadioLink, I requested the source code from RadioLink.  I was never provided this source code - this is not a good thing.  It makes people much less inclined to assist with your future products (and has lead to a large warning notice on our Wiki to steer clear of your products).

n.b. I only checked this far enough to note you hadn't reserved the board ID, so I did that for you.

peterbarker commented:
... further to the above, this came to our Development Call: https://github.com/ArduPilot/ardupilot_wiki/pull/5625 .  That PR was created 3 months ago.

I did manage to convince people to hold off merging that PR for 2 weeks.

*Please* speak to your executives.  The source code has to be released for the MiniPix, and your "secret code" is *not* worth antagonising this project.

peterbarker commented:
Revisitting this in 2 weeks.

Hwurzburg reviewed (changes requested):
needs README.md and images following this format and providing all the same sections and info https://github.com/ArduPilot/ardupilot/tree/master/libraries/AP_HAL_ChibiOS/hwdef/SpeedyBeeF405WING

peterbarker commented:
... actually giving this another week in deference to Chinese New Year.

@radiolinkW - please have a look at that PR I referenced above (https://github.com/ArduPilot/ardupilot_wiki/pull/5625), and the Wiki page for the MiniPix (https://ardupilot.org/copter/docs/common-radiolink-minipix.html).  RadioLink is the *only* vendor that is getting this treatment - developers are unhappy about the GPLv3 license violation.

We'll still merge *this* PR once it has met with Henry's approval, but please do understand the issues we're having with the RadioLink organisation.

... also note that Henry has requested more information in this PR.

radiolinkW commented:
> I've created a PR to reserve the board ID you're using in here.
> 
> Please note that RadioLink still owe me access to source code per ArduPilot's GPLv3.
> 
> Having bought a MiniPix, running custom firmware from RadioLink, I requested the source code from RadioLink. I was never provided this source code - this is not a good thing. It makes people much less inclined to assist with your future products (and has lead to a large warning notice on our Wiki to steer clear of your products).
> 
> n.b. I only checked this far enough to note you hadn't reserved the board ID, so I did that for you.

Unfortunately, Our developers have also communicated with you before, and due to non-standard code commit and other internal reasons within the company, our custom code has not never publicly released,

The code is now publicly available[ here](https://github.com/radiolinkW/ardupilot/tree/MiniPix).

It should be noted that our custom code is based on NuttX . After transferring to ChibiOS, our minipix firmware has been updated with the Ardupilot firmware, so it has not been updated,

And it is also recommended to download the official firmware of Ardupilot for MiniPix.

radiolinkW commented:
> ... further to the above, this came to our Development Call: [ArduPilot/ardupilot_wiki#5625](https://github.com/ArduPilot/ardupilot_wiki/pull/5625) . That PR was created 3 months ago.
> 
> I did manage to convince people to hold off merging that PR for 2 weeks.
> 
> _Please_ speak to your executives. The source code has to be released for the MiniPix, and your "secret code" is _not_ worth antagonising this project.

Thank you very much for your patience. Due to the Chinese New Year holiday, I only saw the news now. The custom code for MiniPix has been released now.

radiolinkW commented:
> ... actually giving this another week in deference to Chinese New Year.
> 
> @radiolinkW - please have a look at that PR I referenced above ([ArduPilot/ardupilot_wiki#5625](https://github.com/ArduPilot/ardupilot_wiki/pull/5625)), and the Wiki page for the MiniPix (https://ardupilot.org/copter/docs/common-radiolink-minipix.html). RadioLink is the _only_ vendor that is getting this treatment - developers are unhappy about the GPLv3 license violation.
> 
> We'll still merge _this_ PR once it has met with Henry's approval, but please do understand the issues we're having with the RadioLink organisation.
> 
> ... also note that Henry has requested more information in this PR.



> ... actually giving this another week in deference to Chinese New Year.
> 
> @radiolinkW - please have a look at that PR I referenced above ([ArduPilot/ardupilot_wiki#5625](https://github.com/ArduPilot/ardupilot_wiki/pull/5625)), and the Wiki page for the MiniPix (https://ardupilot.org/copter/docs/common-radiolink-minipix.html). RadioLink is the _only_ vendor that is getting this treatment - developers are unhappy about the GPLv3 license violation.
> 
> We'll still merge _this_ PR once it has met with Henry's approval, but please do understand the issues we're having with the RadioLink organisation.
> 
> ... also note that Henry has requested more information in this PR.

Thank you for agreeing to merge this PR, I will prepare the readme file and images later.

radiolinkW commented:
> ... actually giving this another week in deference to Chinese New Year.
> 
> @radiolinkW - please have a look at that PR I referenced above ([ArduPilot/ardupilot_wiki#5625](https://github.com/ArduPilot/ardupilot_wiki/pull/5625)), and the Wiki page for the MiniPix (https://ardupilot.org/copter/docs/common-radiolink-minipix.html). RadioLink is the _only_ vendor that is getting this treatment - developers are unhappy about the GPLv3 license violation.
> 
> We'll still merge _this_ PR once it has met with Henry's approval, but please do understand the issues we're having with the RadioLink organisation.
> 
> ... also note that Henry has requested more information in this PR.

Thank you for agreeing to merge this PR, I will prepare the readme file and images later.

radiolinkW commented:
@Hwurzburg @peterbarker  The readme and images has been committed.

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/RadiolinkPIX6/Readme.md:
```diff
@@ -0,0 +1,296 @@
+## UART Mapping
+
+- SERIAL0 -> USB
+- SERIAL1 -> USART2 (Telem1) RTS/CTS pins
```
```suggestion
- SERIAL1 -> USART2 (Telem1) RTS/CTS pins ,RX DMA capable
```

khancyr commented:
I am glad we are gooing forward in the right direction and welcome you into following the project rules and respect of the project license.
As a side note, the GPLv3 don't require you to put your code online or anyway publicly. But you need to provide it to your customer that ask for the source code !  That is what Peter asked.

peterbarker commented:
> The code is now publicly available[ here](https://github.com/radiolinkW/ardupilot/tree/MiniPix).

Fantastic news, thanks!

radiolinkW commented on libraries/AP_HAL_ChibiOS/hwdef/RadiolinkPIX6/Readme.md:
```diff
@@ -0,0 +1,296 @@
+- SERIAL0 -> USB
+- SERIAL1 -> USART2 (Telem1) RTS/CTS pins
+- SERIAL2 -> USART3 (Telem2) RTS/CTS pins
+- SERIAL3 -> USART1 (GPS1)
```
Updated commit

radiolinkW commented:
@peterbarker @khancyr 
Why is your response so slow?

How long can this PR be merged?

Hwurzburg commented:
this PR still needs to be squashed and split into two library commits... I attempted to do this for you, but you havent given permissions to allow me to push to this  repo branch....so I created another PR #26388....we can merge that or you can fix this one

tridge commented:
@radiolinkW the code which RadioLink had not released was for the ChibiOS builds for the minipix not the ancient nuttx code you linked to here

the apj files on this website do not have source code. The same with the MissionPlanner MSI file.

https://www.radiolink.com/minipix_opensource_firmware

there are versions of the minipix which use single flash page storage which is not supported by ArduPilot but you provide apj files for that one your website

radiolinkW commented:
> this PR still needs to be squashed and split into two library commits... I attempted to do this for you, but you havent given permissions to allow me to push to this repo branch....so I created another PR #26388....we can merge that or you can fix this one

It has been squashed and split into two commits.

radiolinkW commented:
> @radiolinkW the code which RadioLink had not released was for the ChibiOS builds for the minipix not the ancient nuttx code you linked to here
> 
> the apj files on this website do not have source code. The same with the MissionPlanner MSI file.
> 
> https://www.radiolink.com/minipix_opensource_firmware
> 
> there are versions of the minipix which use single flash page storage which is not supported by ArduPilot but you provide apj files for that one your website

@tridge  the code which use single flash page storage is publicly available [here](https://github.com/radiolinkW/ardupilot/tree/MiniPix-v1.1).

This only adds a single page storage parameters, nothing else has been changed.

radiolinkW commented:
@Hwurzburg @peterbarker Is there any problem with this PR? Why hasn't it been merged for so long.

radiolinkW commented:
> this PR still needs to be squashed and split into two library commits... I attempted to do this for you, but you havent given permissions to allow me to push to this repo branch....so I created another PR #26388....we can merge that or you can fix this one

You can now push your commits to that branch.

tridge commented:
> @tridge the code which use single flash page storage is publicly available [here](https://github.com/radiolinkW/ardupilot/tree/MiniPix-v1.1).

thanks, we really should move this to being a 
```
define AP_STORAGE_FLASH_SINGLE_PAGE 1
```
and then add support for that in the standard AP_FlashStorage. There will still be users of your MiniPix-v1.1 board out there, and we should have support for that board in our releases so users can get bug fixes and new features.
Can you do a PR to make this an optional build in AP_FlashStorage?

> @Hwurzburg @peterbarker Is there any problem with this PR? Why hasn't it been merged for so long.

now that you've pointed us at the missing code I think we can merge this. We will discuss on the dev call today.

Note that a radiolink engineer previously told me that radiolink was deliberately not releasing the code, and thus deliberately violating our license. That is what led us to be careful with anything from radiolink. We'd be delighted to welcome you if you do start playing by the rules properly.
Have you released all your changes to the MissionPlanner source code? You had a custom MissionPlanner on your website

peterbarker commented:
I've merged this PR.

I think we'd all like to hear an answer to tridge's question on MissionPlanner.  If the fork was simply to allow updating of the MiniPix firmware from your own resources then we might be able to add some form of structure to MissionPlanner to allow for that.  Better not to have that problem in the first place - "upstream first", so your boards continue to work and improve as ArduPilot does.

radiolinkW commented:
> > @tridge the code which use single flash page storage is publicly available [here](https://github.com/radiolinkW/ardupilot/tree/MiniPix-v1.1).
> 
> thanks, we really should move this to being a
> 
> ```
> define AP_STORAGE_FLASH_SINGLE_PAGE 1
> ```
> 
> and then add support for that in the standard AP_FlashStorage. There will still be users of your MiniPix-v1.1 board out there, and we should have support for that board in our releases so users can get bug fixes and new features. Can you do a PR to make this an optional build in AP_FlashStorage?
> 
> > @Hwurzburg @peterbarker Is there any problem with this PR? Why hasn't it been merged for so long.
> 
> now that you've pointed us at the missing code I think we can merge this. We will discuss on the dev call today.
> 
> Note that a radiolink engineer previously told me that radiolink was deliberately not releasing the code, and thus deliberately violating our license. That is what led us to be careful with anything from radiolink. We'd be delighted to welcome you if you do start playing by the rules properly. Have you released all your changes to the MissionPlanner source code? You had a custom MissionPlanner on your website

I‘m really sorry, I only have time to reply to you now.

All changes to our custom MissionPlanner have been released [here](https://github.com/radiolinkW/MissionPlanner.git).

Please note that due to the previous commits being difficult to read,it has been reorganized into a standard commit.

radiolinkW commented:
> I've merged this PR.
> 
> I think we'd all like to hear an answer to tridge's question on MissionPlanner. If the fork was simply to allow updating of the MiniPix firmware from your own resources then we might be able to add some form of structure to MissionPlanner to allow for that. Better not to have that problem in the first place - "upstream first", so your boards continue to work and improve as ArduPilot does.

Yes, it was simply to allow updating of the MiniPix firmware from own resources.

@peterbarker Thank you very much for your merger.but why hasn't been merged into release branchs such as Copter？
